src/components/VisualizationPlugin/__tests__/VisualizationPlugin.spec.js

```diff
@@ -170,7 +174,9 @@ describe('VisualizationPlugin', () => {
             // eslint-disable-next-line no-import-assign, import/namespace
             api.apiFetchAnalytics = jest.fn().mockRejectedValue('error')
 
+            const originalConsoleError = console.error
             await canvas()
+            console.error = originalConsoleError
 
             expect(defaultProps.onError).toHaveBeenCalled()
```

Comment:
Perhaps there is a more elegant way to achieve this, but I think this is probably acceptable here:
- We do in fact want an error to happen here
- We don't want to clutter the test output
- We want to assert that `defaultProps.onError` is called

Reply:
I had a go at testing this in a more conventional way:
```js
        it('calls onError callback when an exception is thrown', async () => {
            // eslint-disable-next-line no-import-assign, import/namespace
            api.apiFetchAnalytics = jest.fn().mockRejectedValue('error')

            expect(async () => {
                await canvas()
            }).rejects.toThrow()

            expect(defaultProps.onError).toHaveBeenCalled()
        })
 ```
 
But that doesn't work because the component does actually catch the error, so the test fails with the following message from Jest:
 ```
     expect(received).rejects.toThrow()

    Received promise resolved instead of rejected
    Resolved to value: undefined
 ```
 
 So I think that probably the solution I have above is correct. Even though it looks a bit hacky.